src/lib/mina_base/zkapp_precondition.ml

```diff
@@ -523,6 +523,18 @@ module Account = struct
 
   let is_accept : t -> bool = equal accept
 
+  let nonce (n : Account.Nonce.t) =
+    let nonce : _ Numeric.t = Check { lower = n; upper = n } in
+    { accept with nonce }
+
+  let is_nonce (t : t) =
```

Comment:
This helper function seems misleading, and in the context where it is used (account_update.ml:1449), the logic is no longer the same. In order to check if a precondition is equivalent to the old `Nonce` precondition helper, you need to check that all other preconditions are set to `Ignore`.

Reply:
I think that's what `is_accept { t with nonce = Ignore }` is doing, yes?

Reply:
Yes, you are correct; that wasn't obvious to me when I read the code. Maybe this deserves a very brief comment?